conninfo/src/conninfo-v1.0-wd01.md

```diff
@@ -0,0 +1,238 @@
+###### [RFC2119]
+Bradner, S., "Key words for use in RFCs to Indicate Requirement Levels", BCP 14, RFC 2119, DOI 10.17487/RFC2119, March 1997, http://www.rfc-editor.org/info/rfc2119.
+###### [RFC2616]
+Fielding, R., Gettys, J., Mogul, J., Frystyk, H., Masinter, L., Leach, P., and T. Berners-Lee, "Hypertext Transfer Protocol -- HTTP/1.1", RFC 2616, DOI 10.17487/RFC2616, June 1999, <https://www.rfc-editor.org/info/rfc2616>.
+###### [RFC8174]
+Leiba, B., "Ambiguity of Uppercase vs Lowercase in RFC 2119 Key Words", BCP 14, RFC 8174, DOI 10.17487/RFC8174, May 2017, http://www.rfc-editor.org/info/rfc8174.
+
+## 1.4 Non-Normative References
+
+###### [RFC3552]
+Rescorla, E. and B. Korver, "Guidelines for Writing RFC Text on Security Considerations", BCP 72, RFC 3552, DOI 10.17487/RFC3552, July 2003, https://www.rfc-editor.org/info/rfc3552.
+
+-------
+
+# 2 Overview
+
+This specification recommends a standard way for an AMQP container to reveal information about
+itself and its operating environment to its partner.   This allows a container to expose information
+that would be useful for problem diagnosis, such as process identifiers and software version
+information.
+
+This specification is equally appropriate for AMQP container fulfilling the role of clients or
+servers, or containers supporting a peer-to-peer scenario.
+
+-------
+
+# 3 Definitions
+
+An AMQP host that implements this specification MUST provide the following entry in the properties
+field of the AMQP `open` performative [[AMQP-v1.0](#AMQP-v1.0)]. The key of the entry MUST be the symbol
+`connection-info`. The value of the entry MUST be of type `fields`.
+
+The following key values have defined meaning. An implementation MAY choose to omit a key from the map.
+An implementation MAY also include keys of its own. If an implementation receives a key from its partner
+that it does not recognise, it SHOULD ignore it.
+
+# 3.1 connection-info
+
+`<type name="connection-info" class="restricted" source="fields"/>`
+
+The content of `connection-info` allows the container to provide information about itself and its operating
+environment to its partner.
+
+**Table 1-1. connection-info**
+
+| Name | Description |
+| :--- | :--- |
+| `process-identifier` | The value of the process-identifier MUST be an AMQP string. The purpose of the value is to identify the process running this AMQP host. It is RECOMMENDED that this is an operating system PID. |
+| `network-host` | The value of network-host MUST be an AMQP string. The purpose of the value is to identify the host running the AMQP container in a way that assists those fault finding within a system. This value MAY be a hostname, fully qualified hostname or IP address. The value MAY be network resolvable by the AMQP peer but the peer MUST NOT rely on this. |
```

Comment:
Since we can have containers inside of VMs sharing the same network interface, good identifiers for process and host may be more complicated than a a PID, so I'm not sure we should go as far as recommending a particular construct here. Also, when you come through a a load balancer, the network-host as seen by the client is different from the backend that it actually talking back to the client and a network address may not be useful as you state, but it may also be harmful to disclose that detail. 

I would merge these two properties together and leave it completely up to the implementer to provide a good value for whatever the process/host is. I assume the key reason for having this is diagnostics and what matters there is for the client to get some sort of reference value that the other side can make sense of.

Reply:
By leaving it completely up to the implementer we lose a lot of value here.  A common use case here is being able to identify clients connecting into a service/broker.  If every client library chose its own form of identifier here, it rather defeats the point

Reply:
My motivation for the inclusion of network-host/process-identifier was to aid faulting finding in complex systems.  I am being influenced by experience of watching operations teams (responsible for a broker) struggle to identify the source of a client producing problematic traffic.

I agree that in a containerised world, a process-identifier value may be worthless (likely to be 1).   For a network-host, at least, in a Kubernetes container, I think this would tend to be a pod name., which would have diagnostic value. 

The implementors of this specification are likely going to be authors of AMQP libraries who won't know the library's target environment(s).   They are likely to assume a traditional/bare metal common denominator. I would expect assignments such as:
*  `process-identifier` to the result calls like `Process.GetCurrentProcess().Id.ToString()` / `sprintf(buf,"%d", getpid())`
* `network-host` `System.Net.Dns.GetHostName()`/ `gethostname(buf, len)`